### DISCUSSION TITLE
`focus-follows-mouse` conflicts with keyboard split navigation

### DISCUSSION THREAD

**Discussion by weirdan** [ghostty-org/ghostty#3865] [Feature Requests, Ideas] [answered]
It appears that `focus-follow-mouse` is incompatible with navigating between spans using the keyboard. In the attached video the mouse is over the bottom split and I'm pressing `Ctrl+Alt+up` repeatedly to switch to the top split. The focus moves up for a moment, but then switches back to the split under the mouse cursor (bottom).

It may seem as if those are mutually exclusive, but it works in Sway, which also uses 'focus follows mouse' approach, but with a twist. There, for the focus to move from window A to window B the mouse pointer has to _move_ from window A to window B. If, however, the focus is on the window A but the mouse pointer is over window B the focus does not switch until you move the mouse over window A and then to window B.

I suggest ghostty follows the approach Sway uses.

[recording.webm](https://github.com/user-attachments/assets/f46ce28f-5c5a-49c2-b675-b4d1230c0fe3)

<details>
<summary>`ghostty +version`</summary>

```
Ghostty 1.0.1-main+5293fc9c

Version
  - version: 1.0.1-main+5293fc9c
  - channel: tip
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.gtk
  - font engine: font.main.Backend.fontconfig_freetype
  - renderer   : renderer.OpenGL
  - libxev     : main.Backend.io_uring
  - GTK version:
    build      : 4.16.7
    runtime    : 4.16.7
  - libadwaita : enabled
    build      : 1.6.2
    runtime    : 1.6.2
  - libX11     : enabled
```
</details>

**Comment by mitchellh** [comment, marked as answer]
This was fixed in 1.0.1
